sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/change_feed_utils.py

```diff
@@ -0,0 +1,151 @@
+    :type is_start_from_beginning: bool
+    :keyword start_time: Must be in supported types.
+    :type start_time: Union[~datetime.datetime, Literal["Now", "Beginning"]]
+    :type kwargs: dict[str, Any]
+    """
+    # Filter items with value None
+    kwargs = {key: value for key, value in kwargs.items() if value is not None}
+
+    # Validate the keyword arguments
+    if "change_feed_mode" in kwargs:
+        change_feed_mode = kwargs["change_feed_mode"]
+        if change_feed_mode not in ChangeFeedMode:
+            raise ValueError(
+                f"Invalid change_feed_mode was used: '{kwargs['change_feed_mode']}'."
+                f" Supported 'change_feed_modes' are {ChangeFeedMode.to_string()}.")
+
+        if change_feed_mode == ChangeFeedMode.ALL_VERSIONS_AND_DELETES:
+            if "partition_key_range_id" in kwargs:
+                raise ValueError(
+                    f"'{ChangeFeedMode.ALL_VERSIONS_AND_DELETES}' mode is not supported if 'partition_key_range_id'"
+                    f" was used. Please use 'feed_range' instead.")
+            if "is_start_from_beginning" in kwargs and kwargs["is_start_from_beginning"] is not False:
+                raise ValueError(
+                    f"'{ChangeFeedMode.ALL_VERSIONS_AND_DELETES}' mode is only supported if 'is_start_from_beginning'"
+                    f" is 'False'. Please use 'is_start_from_beginning=False' or 'continuation' instead.")
+            if "start_time" in kwargs and kwargs["start_time"] != "Now":
+                raise ValueError(
+                    f"'{ChangeFeedMode.ALL_VERSIONS_AND_DELETES}' mode is only supports if 'start_time' is 'Now'."
+                    f" Please use 'start_time=\"Now\"' or 'continuation' instead.")
+
+    if "partition_key_range_id" in kwargs:
+        warnings.warn(
+            "'partition_key_range_id' is deprecated. Please pass in 'feed_range' instead.",
+            DeprecationWarning
+        )
+
+    if "is_start_from_beginning" in kwargs:
+        warnings.warn(
+            "'is_start_from_beginning' is deprecated. Please pass in 'start_time' instead.",
+            DeprecationWarning
+        )
+
+        if "start_time" in kwargs:
+            raise ValueError("'is_start_from_beginning' and 'start_time' are exclusive, please only set one of them.")
+
+        if not isinstance(kwargs["is_start_from_beginning"], bool):
+            raise TypeError(
+                f"'is_start_from_beginning' must be 'bool' type,"
+                f" but given '{type(kwargs['is_start_from_beginning']).__name__}'.")
+
+    elif "start_time" in kwargs:
+        start_time = kwargs['start_time']
+        if not isinstance(start_time, datetime) and not isinstance(start_time, str):
+            raise TypeError(
+                f"'start_time' must be either a 'datetime' or 'str' type, but given '{type(start_time).__name__}'.")
+
+        if isinstance(start_time, str) and start_time not in ["Now", "Beginning"]:
```

Comment:
I would make this check case-insensitive.
```suggestion
        if isinstance(start_time, str) and start_time.lower() not in ["now", "beginning"]:
```